src/com/goide/inspections/GoInfiniteForInspection.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2013-2016 Sergey Ignatov, Alexander Zolotov, Florin Patan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.goide.inspections;
+
+import com.goide.psi.*;
+import com.goide.quickfix.GoReplaceWithSelectStatementQuickFix;
+import com.intellij.codeInspection.LocalInspectionToolSession;
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import org.jetbrains.annotations.NotNull;
+
+public class GoInfiniteForInspection extends GoInspectionBase {
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull final ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+      @Override
+      public void visitForStatement(@NotNull GoForStatement o) {
+        super.visitForStatement(o);
+
+        GoBlock forBlock = o.getBlock();
+        GoRangeClause forRangeClause = o.getRangeClause();
+        GoForClause forClause = o.getForClause();
+        if (o.getExpression() != null ||
```

Comment:
please invert the condition. this is really strange to consider all loops as infinite *except* `o.getExpression() != null` or block == nul or ... etc